packages/widgets/src/lib/components/InferenceWidget/shared/WidgetFileInput/WidgetFileInput.svelte

```diff
@@ -21,36 +29,45 @@
 </script>
 
 {#if !isDisabled}
-	<div
-		class={classNames}
-		on:dragenter={() => {
-			isDragging = true;
-		}}
-		on:dragover|preventDefault
-		on:dragleave={() => {
-			isDragging = false;
-		}}
-		on:drop|preventDefault={(e) => {
-			isDragging = false;
-			fileInput.files = e.dataTransfer?.files ?? null;
-			onChange();
-		}}
-	>
-		<label class="btn-widget {isDragging ? 'ring' : ''} {isLoading ? 'text-gray-600' : ''}">
-			{#if isLoading}
-				<IconSpin classNames="-ml-1 mr-1.5 text-gray-600 animate-spin" />
-			{:else}
-				<IconFile classNames="-ml-1 mr-1.5" />
-			{/if}
-			<input
-				{accept}
-				bind:this={fileInput}
-				on:change={onChange}
-				disabled={isLoading || isDisabled}
-				style="display: none;"
-				type="file"
-			/>
-			{label}
-		</label>
-	</div>
+	<LogInPopover bind:open={popOverOpen}>
+		<button
+			class={classNames}
+			on:click={(e) => {
+				if (!$isLoggedIn) {
+					popOverOpen = true;
+					e.preventDefault();
+					return;
+				}
+			}}
```

Comment:
(nit) maybe cleaner ? wdyt @krampstudio
```suggestion
			on:click|preventDefault={(e) => {
				if (!$isLoggedIn) {
					popOverOpen = true;
					return;
				}
			}}
```

Reply:
I think we want `preventDefault` only if `!$isLoggedIn`. Otherwise, we want the default behaviour